classes/local/v1p2/collections/classes_for_user.php

```diff
@@ -0,0 +1,49 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * One Roster Enrolment Client.
+ *
+ * @package    enrol_oneroster
+ * @copyright  Andrew Nicols <andrew@nicols.co.uk>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace enrol_oneroster\local\v1p1\collections;
+
+use enrol_oneroster\local\interfaces\container as container_interface;
+use enrol_oneroster\local\v1p1\endpoints\rostering as rostering_endpoint;
+use enrol_oneroster\local\collections\classes as classes_collection;
+
+/**
+ * One Roster Courses collection.
+ *
+ * @package    enrol_oneroster
+ * @copyright  Andrew Nicols <andrew@nicols.co.uk>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class classes_for_user extends classes_collection {
+
+    /**
+     * Get the operation ID for the endpoint, otherwise known as the name of the endpoint.
+     *
+     * @param   container_interface $container The container to which this entity belongs
+     * @return  string
+     */
+    protected static function get_operation_id(container_interface $container): string {
+        return rostering_endpoint::getClassesForUser;
```

Comment:
This is a strange one. its calling a static invoation of the Method name? This feels like it would be much better as a constant somewhere. This is buggy, changing a method name would have side effects that would be strange.